mix.exs

```diff
@@ -73,4 +73,27 @@ defmodule VintageNetQMI.MixProject do
       links: %{"GitHub" => @source_url}
     ]
   end
+
+  # utility function that strips the original csv
+  # database file down to the fields we need. 
+  @doc false
+  def build_csv(path) do
+    csv_data =
+      path
+      |> File.read!()
+      |> String.trim()
+      |> String.split("\r\n")
+      |> Enum.map(&String.split(&1, ";"))
+      |> tl()
+      |> Enum.reject(fn [_, _, _, _, _, _, _, pro, _, _ | _] ->
+        String.length(pro) == 0
+      end)
+      |> Enum.map(fn [mcc, mnc, _, _, _, _, _, pro, _, _ | _] -> [mcc, mnc, pro] end)
+      |> Enum.uniq_by(fn [mcc, mnc, _] -> {mcc, mnc} end)
+      |> Enum.map(&Enum.join(&1, ";"))
+      |> Enum.join("\n")
```

Comment:
Did you not like using `Stream` like in the other code? I thought it was nice that it split the lines for you.

Reply:
whoops, the Stream version is good. updated.